app/controllers/login.ts

```diff
@@ -0,0 +1,54 @@
+'use strict';
+export default function(app) {
+  app.controller('LoginCtrl', ['$scope', '$location', 'authService', '$http', 'config',
+    function($scope, $location, authService, $http, config) {
+
+      $scope.auth = authService;
+
+      $scope.login = {
+        emailOrUsername: '',
+        password: ''
+      };
+
+      $scope.hasError = function(field) {
+        const form = $scope.loginForm;
+        return (form.$submitted || form[field].$touched) &&
+          form[field].$invalid;
+      };
+
+      $scope.$watch('emailOrUsername', function() {
+        $scope.emailError = undefined;
+        $scope.responseError = undefined;
+      });
```

Comment:
Do we really need these `$watch` statements? `emailError` is not set anywhere so it seems to be safe to remove (also from the html). `responseError` is be set to undefined in `login()` and this seems to be sufficient.